src/script/backup/zipWorker.ts

```diff
@@ -46,6 +71,45 @@ export async function handleZipEvent(payload: Payload) {
   }
 }
 
+// Encrypt a file
+async function encryptFile(fileContent: Uint8Array, encryptionKey: Uint8Array): Promise<Uint8Array> {
+  await sodium.ready;
+
+  const headerBytes = sodium.crypto_secretstream_xchacha20poly1305_HEADERBYTES;
+  const stateAndHeader = sodium.crypto_secretstream_xchacha20poly1305_init_push(encryptionKey);
+  const {state, header} = stateAndHeader;
+
+  const encryptedFile = sodium.crypto_secretstream_xchacha20poly1305_push(
+    state,
+    fileContent,
+    null,
+    sodium.crypto_secretstream_xchacha20poly1305_TAG_MESSAGE,
+  );
+  const encrypted = new Uint8Array(headerBytes + encryptedFile.length);
+
+  encrypted.set(header);
+  encrypted.set(encryptedFile, headerBytes);
+
+  return encrypted;
+}
+
+// Decrypt a file
+export async function decryptFile(encryptedDataSource: Uint8Array, encryptionKey: Uint8Array, headerLength: number) {
+  await sodium.ready;
+
+  const metaDataHeader = headerLength;
+  const headerBytes = sodium.crypto_secretstream_xchacha20poly1305_HEADERBYTES;
+  const header = encryptedDataSource.slice(metaDataHeader, metaDataHeader + headerBytes);
+  const state = sodium.crypto_secretstream_xchacha20poly1305_init_pull(header, encryptionKey);
+  const encryptedContent = encryptedDataSource.slice(headerBytes + metaDataHeader);
+  const decrypted = sodium.crypto_secretstream_xchacha20poly1305_pull(state, encryptedContent);
+
+  if (!!!decrypted) {
```

Comment:
is this necessary? I think this should also do it:
```suggestion
  if (!decrypted) {
```